### PR TITLE
Dataset and DataCollator for BERT Next Sentence Prediction (NSP) task

### PR DESCRIPTION
**Pull request by mojave-pku**
Add `DataCollatorForNextSencencePrediction` and `TextDatasetForNextSencencePrediction` to support mlm and next sentence prediction objectives together.

### CODE REVIEW

File: src/transformers/data/data_collator.py
Comment (sgugger) [inline src/transformers/data/data_collator.py:334]: Could we support dict as well since the nlp library will return that?

**Review by sgugger — APPROVED** [review]
This looks good to me, just one nit before we can merge.

**Comment by HuangLianzhe** [comment]
Hi, @sgugger ! I add dict inputs support like `DataCollatorForLanguageModeling` according to your suggestion, but now there is a  conflict in `src/transformers/__init__.py`. Do I need to resolve it or leave it to you?

**Comment by sgugger** [comment]
I can take care of the final merge once this is all good and @LysandreJik approved, it's due to a new version of isort.

File: src/transformers/data/data_collator.py
Comment (HuangLianzhe) [reply src/transformers/data/data_collator.py:336]: Sorry, I'm a little confused.
The `labels` you mentioned are nsp/mlm labels, or labels for a specific task?
Since none of data collators in this file grab the `token_type_ids` and `labels`, they just take the examples out of the dict, and do nothing else. 
Thank you~

Comment (sgugger) [reply src/transformers/data/data_collator.py:336]: Ah sorry, I was reading this wrong.

Comment (HuangLianzhe) [reply src/transformers/data/data_collator.py:336]: aha, ok~ :-)

**Comment by LysandreJik** [comment]
Could we add a test for this? I just merged `master` in to make sure it has the latest changes.

**Comment by HuangLianzhe** [comment]
After @LysandreJik  merge the `master` branch, many files need to be reformatted. May I use `make style` to reformat all these files (138 files reformatted) before commit, or submit directly (but the `check_code_quality` will not pass)?

**Comment by HuangLianzhe** [comment]
After @LysandreJik merge the `master` branch, many files need to be reformatted. 
To clearly show the codes I modified, I did not include the changes caused by `make style` of other files in those commits, so `check_code_quality` will not pass.

**Review by LysandreJik — APPROVED** [review]
Cool, thanks for adding the test!

**Pull request merged**
